Tools/AP_Periph/can.cpp

```diff
@@ -902,7 +902,23 @@ static void process1HzTasks(uint64_t timestamp_usec)
 
 #if !defined(HAL_NO_FLASH_SUPPORT) && !defined(HAL_NO_ROMFS_SUPPORT)
     if (periph.g.flash_bootloader.get()) {
+        const uint8_t flash_bl = periph.g.flash_bootloader.get();
         periph.g.flash_bootloader.set_and_save_ifchanged(0);
+        if (flash_bl == 42) {
+            // magic developer value to test watchdog support with main loop lockup
+            while (true) {
+                can_printf("entering lockup\n");
+                hal.scheduler->delay(100);
+            }
+        }
+        if (flash_bl == 43) {
+            // magic developer value to test watchdog support with hard fault
+            can_printf("entering fault\n");
+            void *foo = (void*)0xE000ED38;
```

Comment:
Nitpicking, should we be using a define from ChibiOS here? &SCB->BFAR

Reply:
it's just a value Peter chose (see the same code in GCS_MAVLink.cpp). It happens to be BFAR, because that was the example watchdog he had. It is really just supposed to be a distinctive value